### PR TITLE
:book: Added prior art links to release documentation.

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
Added prior art links to release documentation.

### CODE REVIEW

File: docs/releasing.md [hunk 1]
```diff
@@ -15,12 +15,12 @@ Things you should check before making a release:
   [Metal3 release process](https://github.com/metal3-io/metal3-docs/blob/main/processes/releasing.md)
   for high-level process and possible follow-up actions
 - Verify CAPI go module is uplifted in root, `api/` and `test/` go modules and
-  `cluster-api/test` module in `test/` go module.
+  `cluster-api/test` module in `test/` go module. Prior art: [#1157](https://github.com/metal3-io/cluster-api-provider-metal3/pull/1157)
 - Verify controller Go modules use latest corresponding CAPI modules
```
Comment (tuminoid) [inline docs/releasing.md:19]: Add prior art for controller bumps as well.

**Review by tuminoid — COMMENTED** [review]
Otherwise OK, but for completeness sake, let's add prior art to controller bumps as well.

**Comment by tuminoid** [comment]
/override test-centos-e2e-integration-main 
/override test-ubuntu-integration-main

Irrelevant tests. Save trees.

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Review by kashifest — COMMENTED** [review]
/approve

**Pull request merged**